File: client/web/sudoo/lib/domain/model/order/shipment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sudoo/domain/model/order/shipment_type.dart';

part 'shipment.g.dart';

@JsonSerializable()
class Shipment {
  final ShipmentType shipmentType;
  final int deliveryTime;
  final DateTime? receivedDateTime;
  final double shipmentPrice;

  Shipment(
    this.shipmentType,
    this.deliveryTime,
    this.receivedDateTime,
    this.shipmentPrice,
  );

  factory Shipment.fromJson(Map<String, dynamic> json) => _$ShipmentFromJson(json);

  Map<String, dynamic> toJson() => _$ShipmentToJson(this);
}

File: client/mobile/ShipperDemo/app/src/main/java/com/sudoo/shipment/model/Shipment.kt
package com.sudoo.shipment.model

import java.time.LocalDateTime

data class Shipment(
    val shipmentType: ShipmentType = ShipmentType.STANDARD,
    val deliveryTime: Int = 0,
    val shipmentPrice: Double = 0.0,
    val receivedDateTime: LocalDateTime? = null,
)
